### PR TITLE
imlemented required methods of linked list

### PR DESCRIPTION
**Pull request by tatsiak**
Not sure, but if I understood task right - this methods should be enough to implement stack and queue.

### CODE REVIEW

File: test/linkedList.spec.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            linkedList.pushIntoStart(2);
+            linkedList.pushIntoStart(4);
+            linkedList.pop().should.be.eql(1);
+            linkedList.pop().should.be.eql(2);
+            linkedList.pop().should.be.eql(4);
+
+        })
+    });
+    describe('#pop', function () {
+        it('should return last element', function () {
```
Comment (pavlom0x) [inline test/linkedList.spec.js]: how `#pushIntoEnd` and `#pop` test different from each other except description and number of elements?
do you really test different behavior?

Comment (pavlom0x) [reply test/linkedList.spec.js]: where test with pop form empty list? 
Try to come up with other test cases

**Review by pavlom0x — CHANGES_REQUESTED** [review]

**Comment by pavlom0x** [comment]
Write to your Reviewers they should review you

File: test/linkedList.spec.js (hunk 1 above)
Comment (tatsiak) [reply test/linkedList.spec.js]: fixed both

**Review by pavlom0x — APPROVED** [review]

File: app/linkedList.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    constructor(value, previous = null) {
+        this._value = value;
+        this._previous = previous;
+        this._next = null;
+    }
+}
+class LinkedList {
+    constructor() {
+        this._start = null;
+        this._end = null;
```
Comment (AlexDoh) [inline app/linkedList.js:16]: No comments to logic of the list. But after adding first element both _start and _end became the same, so maybe as improvement, you can keep only _start element as whole list, because if the list consists only one element - there is no reason to keep 2 equal objects.

File: test/linkedList.spec.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+'use strict'
+const algolib = require('../index.js');
+
+describe('#linkedList', function () {
```
Comment (AlexDoh) [inline test/linkedList.spec.js:4]: You can also create a test, which returns null, when all nodes are removed from a list.

**Review by AlexDoh — COMMENTED** [review]
Added two suggestions to implementation and tests.

File: app/linkedList.js (hunk 2 above)
Comment (tatsiak) [reply app/linkedList.js:16]: It's possible, but it'll make logic more complicated.
Setting both start and end required because every non-first element should have _next and _previous.
So if I decide to left only start - I'll have to change logic of all methods without any grow in performance.
That's why I think I should keep it as it is now.

File: test/linkedList.spec.js (hunk 3 above)
Comment (tatsiak) [reply test/linkedList.spec.js:4]: Great idea!
Added one more assertion.

**Review by AlexDoh — APPROVED** [review]
Approved :)

**Comment by algumen** [comment]
Привет, сорри за задержку с ревью. По реализации багов не нашел ). Рекомендовал бы реализовать еще метод, удаляющий с начала листа, типа popFromStart. 
В целом Approved

**Review by algumen — APPROVED** [review]
